lib/system.nim

```diff
@@ -1611,7 +1611,36 @@ when notJSnotNims and defined(nimSeqsV2):
   include "system/seqs_v2"
 
 when not defined(js):
-  proc newSeqUninitialized*[T: SomeNumber](len: Natural): seq[T] =
+  proc newSeqUninit*[T](len: Natural): seq[T] =
+    ## Creates a new sequence of type `seq[T]` with length `len`.
+    ##
+    ## Note that the sequence will be
+    ## uninitialized. After the creation of the sequence you should assign
+    ## entries to the sequence instead of adding them.
+    runnableExamples:
+      # newSeqUninit can be safely used for types, which don't contain
+      # managed memory or have destructors
+      var x = newSeqUninit[int](3)
+      assert len(x) == 3
+      x[0] = 10
+
+      # be cautious to use it with types with managed memory or destructors
+      proc initStringSeq(x: Natural): seq[string] {.nodestroy.} =
+        result = newSeqUninit[string](x)
+        for i in 0..<x: result[i] = "abc"
+
+      let s = initStringSeq(10)
+      assert len(s) == 10
+      assert s[0] == "abc"
+
+    result = newSeqOfCap[T](len)
```

Comment:
missing type constraint for `T`?

Reply:
Nope, it's intended. There is an example to demonstrate how to use it with types having destructors (credits to @araq).

Reply:
"be cautious" is kind of .. unsettling though - ie it feels like a sentence saying that it randomly might or might not work depending on the whims of the compiler, the complexity of the object graph or the mood of the developer

Reply:
also, `nodestroy` on the whole `proc` like that would likely lead to other bugs - ie if you (accidentally) create some temporaries during construction - sounds like a good way to mess up the compiler - not sure it's good to recommend that kind of footgun

Reply:
It's a sharp tool but it fits `newSeqUninit` which is a very sharp tool too. However, we can move these to a new module `unsafeseqs` and then distinguish between `newSeqUninit` which has the constraint `supportsCopyMem(T)` and `newSeqUnsafe` which lacks the constraint.

Reply:
well, that's reasonable along with a better name - this version is not sharp at all beyond its name for plain types ("uninitialised" vs "broken GC with lots of gotchas") 

for the gc case, one would almost need to have a `template newSeqInitIt(size, body): {.nodestroy.}: let it = ...; body` to make it .. reasonable. `nodestroy` is a bit of a shotgun.

Reply:
> we can move these to a new module unsafeseqs and then distinguish between newSeqUninit which has the constraint supportsCopyMem(T) and newSeqUnsafe which lacks the constraint.

done

Reply:
`newSeqUninit` is safe why is it in `unsafeseqs` ?

Reply:
Probably"unsafeseqs" is not a descriptive name. I added both of them there instead of system lest they would break important packages.